### PR TITLE
Support advanced event predicates + multi finger pan

### PR DESCRIPTION
**Pull request by sebakerckhof**
This PR is maybe a bit broad since it implements a couple of things:

It implements the API as specified here:
https://github.com/anvaka/panzoom/issues/40
I'm not sure if we actually need to individually pass propagate and ignore, since they will usually be the same. This API unifieds the different event handlers and adds more of them. It greatly improves the possibilities of this library.

And multi-finger pan, as requested by some people: https://github.com/anvaka/panzoom/issues/63 and https://github.com/anvaka/panzoom/issues/41

This is still a bit of a WIP, it's quite experimental and needs documentation & tests. However I wanted to get it out there for people that were looking for some of these features and to get feedback on the changes so far.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,6 +31,6 @@
     "browserify": "^16.2.3",
     "jsdom": "^11.12.0",
     "tap": "^12.0.0",
-    "uglify-js": "^3.4.9"
+    "terser": "^3.16.1"
```
Comment (sebakerckhof) [inline package.json]: Terser is the successor of uglify-js and it's ES6+ compatible.

File: dist/panzoom.js [hunk 2]
```diff
@@ -56,15 +66,18 @@ function createPanZoom(domElement, options) {
     panController.initTransform(transform)
   }
 
-  var filterKey = typeof options.filterKey === 'function' ? options.filterKey : noop;
```
Comment (sebakerckhof) [inline dist/panzoom.js]: filterKey can be replaced by the 'onKeyDown' predicate

**Comment by Hirbod** [comment]
I tried in incorporate your PR in my local build. After doing so, 2-Finger pinch won't wore anymore. I actually need this: Zoom with 2 Finger pinch, Pan with 2 Finger. Same as google maps does. Did I miss something?

**Comment by anvaka** [comment]
Sorry, this PR caught me in a tough life period and fell through the cracks. At this point the library moved beyond point where merge is feasible. 

I appreciate your contribution and apologize for ignoring it. My fault.

**Pull request closed without merging**